k8s/upgrades/0.8.0-0.8.1/cstor_volume_upgrade.sh

```diff
@@ -0,0 +1,162 @@
+
+function usage() {
+    echo 
+    echo "Usage:"
+    echo 
+    echo "$0 <pv-name> <openebs-namespace>"
+    echo 
+    echo "  <pv-name> Get the PV name using: kubectl get pv"
+    echo "  <openebs-namespace> Get the namespace where openebs"
+    echo "  pods are installed"
+    exit 1
+}
+
+function setDeploymentRecreateStrategy() {
+    dns=$1 # deployment namespace
+    dn=$2  # deployment name
+    currStrategy=`kubectl get deploy -n $dns $dn -o jsonpath="{.spec.strategy.type}"`
+
+    if [ $currStrategy = "RollingUpdate" ]; then
+       kubectl patch deployment --namespace $dns --type json $dn -p "$(cat patch-strategy-recreate.json)"
+       rc=$?; if [ $rc -ne 0 ]; then echo "Upgrade failed | ERROR: $rc"; exit; fi
+       echo "Deployment upgrade strategy set as recreate"
+    else
+       echo "Deployment upgrade strategy was already set as recreate"
+    fi
+}
+
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+pv=$1
+ns=$2
+
+# Check if pv exists
+kubectl get pv $pv &>/dev/null;check_pv=$?
+if [ $check_pv -ne 0 ]; then
+    echo "$pv not found";exit 1;    
+fi
+
+# Check if CASType is cstor
+cas_type=`kubectl get pv $pv -o jsonpath="{.metadata.annotations.openebs\.io/cas-type}"`
+if [ $cas_type != "cstor" ]; then
+    echo "Cstor volume not found";exit 1;
+fi
+
+pvc=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.name}"`
+sc_ns=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+sc_name=`kubectl get pv $pv -o jsonpath="{.spec.storageClassName}"`
+sc_res_ver=`kubectl get sc $sc_name -n $sc_ns -o jsonpath="{.metadata.resourceVersion}"`
+
+################################################################# 
+# STEP: Generate deploy, replicaset and container names from PV #
+#                                                               #
+# NOTES: Ex: If PV="pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc",  #
+#                                                               #
+# c-dep: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-target        # 
+#################################################################
+
+c_dep=$(echo $pv-target)
+c_svc=$(echo $pv)
+c_vol=$(echo $pv)
+
+# Check if openebs resources exist and provisioned version is 0.8
+
+kubectl get deployment $c_dep -n $ns &>/dev/null
+rc=$?; if [ $rc -ne 0 ]; then echo "Target deployment not found: $rc"; exit; fi
+
+openebs_version=`kubectl get deployment $c_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [ $openebs_version != "0.8.0" ]; then
+    echo "Current volumes version is not 0.8.0";exit 1;    
+fi
+
+kubectl get svc $c_svc -n $ns &>/dev/null
+rc=$?; if [ $rc -ne 0 ]; then echo "Service not found: $rc"; exit; fi
+
+
+# Fetch the older target and replica - ReplicaSet objects which need to be 
+# deleted before upgrading. If not deleted, the new pods will be stuck in 
+# creating state - due to affinity rules. 
+
+c_rs=$(kubectl get rs -o name --namespace $ns | grep $c_dep | cut -d '/' -f 2)
```

Comment:
use this instead of 'grep':
$ kubectl get rs -n openebs -l openebs.io/persistent-volume=$pv -o name
replicaset.extensions/pvc-3aecf1d3-1a55-11e9-b63d-42010af002af-target-d7847995b
'cut' is still required